src/dynafed.rs

```diff
@@ -272,3 +369,81 @@ impl Decodable for Params {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use bitcoin;
+    use bitcoin::hashes::hex::ToHex;
+
+    #[test]
+    fn test_param_roots() {
+        // Taken from the following Elements Core test:
+
+        // CScript signblockscript(optocdetype(1));
+        // uint32_t signblock_wl(2);
+        // CScript fp_program(opcodetype(3));
+        // CScript fp_script(opcodetype(4));
+        // std::vector<std::vector<unsigned char>> ext{ {5, 6}, {7} };
+        // 
+        // DynaFedParamEntry compact_entry = DynaFedParamEntry(signblockscript, signblock_wl);
+        // BOOST_CHECK_EQUAL(
+        //     compact_entry.CalculateRoot().GetHex(),
+        //     "4587cf1c2448740c3fe4bf0d3ab22f3f193b18f43c4ee47da3690c53becb7fa4"
+        // );
+        // 
+        // DynaFedParamEntry full_entry = 
+        //     DynaFedParamEntry(signblockscript, signblock_wl, fp_program, fp_script, ext);
+        // BOOST_CHECK_EQUAL(
+        //     full_entry.CalculateRoot().GetHex(),
+        //     "8cd00fe8336c643279c47a7b76ff5e7188033a93c4fb944d98687a46a1c7ce9a"
+        // );
+        // 
+        // DynaFedParams params = DynaFedParams(compact_entry, full_entry);
+        // BOOST_CHECK_EQUAL(
+        //     params.CalculateRoot().GetHex(),
+        //     "ab01e55beb8a8c02d1c6940ff1b2dda1116475e80459856ddc900e89f4a6b68b"
+        // );
+
+        let signblockscript: bitcoin::Script = vec![1].into();
+        let signblock_wl = 2;
+        let fp_program: bitcoin::Script = vec![3].into();
+        let fp_script = vec![4];
+        let ext = vec![vec![5, 6], vec![7]];
+
+        let compact_entry = Params::Compact {
+            signblockscript: signblockscript.clone(),
+            signblock_witness_limit: signblock_wl,
+        };
+        assert_eq!(
+            compact_entry.calculate_root().to_hex(),
+            "dff5f3793abc06a6d75e80fe3cfd47406f732fa4ec9305960ae2a229222a1ad5"
```

Comment:
This doesn't appear to match elements test hash `4587cf1c2448740c3fe4bf0d3ab22f3f193b18f43c4ee47da3690c53becb7fa4`?

Reply:
Ah yeah I forgot to update those after I changed the test. In this PR you can see the correct ones: https://github.com/ElementsProject/elements/pull/719/files
I'll fix this in a force-push.